### PR TITLE
system: add warning when running rootless on cgroupv1

### PR DESCRIPTION
**Pull request by giuseppe**
when running rootless on cgroupv1, Podman+OCI runtime do not setup a separate cgroup for the container.  Spawned containers will run in the same cgroup as the podman system service process, thus will be killed when the cgroup is destroyed, as it is the case when running as a systemd service.

[NO NEW TESTS NEEDED]

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=2170444

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by edsantiago** [comment]
Quick note: this is likely to cause yet more test breakage when we enable cgroupsv1 tests in Debian (and/or in RHEL gating tests). I don't know which tests will fail, but new warnings == new test failures.

File: cmd/podman/system/service_abi.go [hunk 1]
```diff
@@ -106,6 +108,13 @@ func restService(flags *pflag.FlagSet, cfg *entities.PodmanConfig, opts entities
 	// Close the fd right away to not leak it during the entire time of the service.
 	devNullfile.Close()
 
+	cgroupv2, _ := cgroups.IsCgroup2UnifiedMode()
+	fmt.Printf("rootless.IsRootless() && !cgroupv2 %v %v\n", rootless.IsRootless(), cgroupv2)
```
Comment (Luap99) [inline cmd/podman/system/service_abi.go:112]: debug output?

**Comment by edsantiago** [comment]
I am sorry, I know this is rude: I am adding a do-not-merge label. #17305 is really, really close to merging. @cevich has labored many many hours on it over many weeks. I would like that one to merge first, then this one (17589, the warning-message) to rebase and deal with the fallout from the warning message. It is not fair to impose that burden on Chris, or worse on some poor random person who gets the union of both PRs without that conflict resolution. Thank you for understanding.

**Comment by cevich** [comment]
What are you talking about? 17305 was no effort at all.  Passed CI moments after I first pushed it!

Thanks Ed, I'm a bit bias, but I agree :grin: 17305 has been a proper epic PR with a distro-switch and dredging up of so many long-untested pathways.  It was bound to be the nightmare and I'm not surprised conflicts were quick to arise.

**Comment by edsantiago** [comment]
@giuseppe, #17305 has merged, please rebase and repush when convenient. Thanks for your patience.

File: cmd/podman/system/service_abi.go (hunk 1 above)
Comment (giuseppe) [reply cmd/podman/system/service_abi.go]: yes sorry for leaving that, dropped now.

**Comment by giuseppe** [comment]
> I am sorry, I know this is rude: I am adding a do-not-merge label. #17305 is really, really close to merging. @cevich has labored many many hours on it over many weeks. I would like that one to merge first, then this one (17589, the warning-message) to rebase and deal with the fallout from the warning message. It is not fair to impose that burden on Chris, or worse on some poor random person who gets the union of both PRs without that conflict resolution. Thank you for understanding.

no that is completely fine.  Thanks for blocking it and not causing more issues to the long-standing PR :-)

rebased and pushed a new version

**Comment by rhatdan** [comment]
LGTM

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by edsantiago** [comment]
Holds canceled. Thank you!

**Pull request merged**